wrapanapi/systems/rhevm.py

```diff
@@ -1079,3 +1082,15 @@ def import_template(self, edomain, sdomain, cluster, temp_template):
             cluster=types.Cluster(id=cluster_id),
             template=types.Template(id=export_template.id)
         )
+
+    @property
+    def _vnic_profile_service(self):
+        return self.api.system_service().vnic_profiles_service()
+
+    def list_vnic_profiles(self):
+        return self._vnic_profile_service.list()
+
+    @property
+    def vnic_profiles(self):
```

Comment:
Consider a different pattern here - a `RHEVMSystem.get_vnic_profile(profile_name)` method that pulls the try/except currently in `_nic_action` into a common place. That way methods that provide an interface taking a vnic_profile (name or object) can lookup the name they're passed, instead of having to reference and check it.